shared/message/input/PS3Controller.proto

```diff
@@ -0,0 +1,246 @@
+/*
+ * This file is part of the NUbots Codebase.
+ *
+ * The NUbots Codebase is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The NUbots Codebase is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with the NUbots Codebase.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright 2013 NUbots <nubots@nubots.net>
+ */
+
+syntax = "proto3";
+
+package message.input.PS3Controller;
+
+import "google/protobuf/timestamp.proto";
+
+message CrossButton {
+    // The timestamp of the event
+    google.protobuf.Timestamp timestamp = 1;
```

Comment:
If you’re going to include these timestamps you need to make sure the clocks make sense. Otherwise when future person does now - timestamp it’s going to give nonsense

Reply:
then tell me what the clock is

Reply:
Who knows! either some c clock, or maybe a clock on the device itself